lib/util/host-rules.ts

```diff
@@ -9,37 +9,7 @@ import { parseUrl, validateUrl } from './url';
 
 let hostRules: HostRule[] = [];
 
-interface LegacyHostRule {
-  hostName?: string;
-  domainName?: string;
-  baseUrl?: string;
-}
-
-function migrateRule(rule: LegacyHostRule & HostRule): HostRule {
-  const cloned: LegacyHostRule & HostRule = clone(rule);
-  delete cloned.hostName;
-  delete cloned.domainName;
-  delete cloned.baseUrl;
-  const result: HostRule = cloned;
-
-  const { matchHost } = result;
-  const { hostName, domainName, baseUrl } = rule;
-  const hostValues = [matchHost, hostName, domainName, baseUrl].filter(Boolean);
-  if (hostValues.length === 1) {
-    const [matchHost] = hostValues;
-    result.matchHost = matchHost;
-  } else if (hostValues.length > 1) {
-    throw new Error(
-      `hostRules cannot contain more than one host-matching field - use "matchHost" only.`
-    );
-  }
-
-  return result;
-}
-
-export function add(params: HostRule): void {
-  const rule = migrateRule(params);
-
+export function add(rule: HostRule): void {
   const confidentialFields: (keyof HostRule)[] = ['password', 'token'];
   if (rule.matchHost) {
     const parsedUrl = parseUrl(rule.matchHost);
```

Comment:
resolved host is set below 👇

Reply:
From what I can collect, the `migrateRule` call before checking `rule.matchHost` will convert endpoint to matchHost ie. resolvedHost should have benn added in the past as well. Will debug tonight to see if I am missing somehting

Reply:
Caused due to the use a top-scope variable `hostRules` from `host-rules.ts` in the `modules/platform/index` file.
Previously we cloned the rule so the `hostRules` var wan't affected but when I removed the `migrateRule` from the `host-rules` the we didn't clone thus directly changed the `hostRules` object. 🙂

Is this something that should be refactored (the use of global vars)?